### PR TITLE
Bump Netty from 4.1.84 to 4.1.87

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- Bump Netty from `4.1.84.Final` to `4.1.87.Final` (release note: https://netty.io/news/2023/01/12/4-1-87-Final.html)
- with 2 CVE （ including 1 in high risk level) fixed in 4.1.86.Final for 4.1.85 and before, `CVE-2022-41915` and `CVE-2022-41881`
 (https://netty.io/news/2022/12/12/4-1-86-Final.html)
- exclude `netty-handler-ssl-ocsp` which is released with `netty-all` since `4.1.86.Final`, as no SSL ocsp related feature used in kyuubi server

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: dev/dependencyList [hunk 1]
```diff
@@ -132,23 +132,24 @@ metrics-core/4.2.8//metrics-core-4.2.8.jar
...
+netty-all/4.1.87.Final//netty-all-4.1.87.Final.jar
+netty-buffer/4.1.87.Final//netty-buffer-4.1.87.Final.jar
+netty-codec-dns/4.1.87.Final//netty-codec-dns-4.1.87.Final.jar
+netty-codec-http/4.1.87.Final//netty-codec-http-4.1.87.Final.jar
+netty-codec-http2/4.1.87.Final//netty-codec-http2-4.1.87.Final.jar
+netty-codec-socks/4.1.87.Final//netty-codec-socks-4.1.87.Final.jar
+netty-codec/4.1.87.Final//netty-codec-4.1.87.Final.jar
+netty-common/4.1.87.Final//netty-common-4.1.87.Final.jar
+netty-handler-proxy/4.1.87.Final//netty-handler-proxy-4.1.87.Final.jar
+netty-handler-ssl-ocsp/4.1.87.Final//netty-handler-ssl-ocsp-4.1.87.Final.jar
```
Comment (pan3793) [inline dev/dependencyList:144]: it's a new jar, is it expected?

Comment (bowenliang123) [reply dev/dependencyList:144]: It's mentioned in PR description. All I know is 1. `netty-handler-ssl-ocsp` package is released from 4.1.86+, and 2. it's included by `netty-all` which is a dependency of kyuubi server module.

Comment (pan3793) [reply dev/dependencyList]: sgtm

Comment (bowenliang123) [reply dev/dependencyList]: OK, addressed and updated the description for exclusion.

**Review by pan3793 — APPROVED** [review]

**Comment by bowenliang123** [comment]
thanks, merged to master.
